lib/spack/docs/analyze.rst

```diff
@@ -0,0 +1,156 @@
+    spec.yaml
+
+This means that you can always find analyzer output in this folder. Sometimes the output
+is a bit redundant - parsed files from the folder above, but it's been added to
+be namespaced as an analyzer output.
+
+-----------------
+Listing Analyzers
+-----------------
+
+If you aren't familiar with Spack's analyzers, you can quickly list those that 
+are available:
+
+.. code-block:: console
+
+    $ spack analyze --list-analyzers
+    ==> install_files                      : install file listing read from install_manifest.json
+    ==> environment_variables              : environment variables parsed from spack-build-env.txt
+    ==> config_args                        : config args loaded from spack-configure-args.txt
+    ==> abigail                            : Application Binary Interface (ABI) features for objects
+
+
+In the above, the first three are fairly simple - parsing metadata files from
+a package install directory to save
+
+-------------------
+Analyzing a Package
+-------------------
+
+The analyze command, akin to install, will accept a package spec to perform
+an analysis for. The package must be installed. Let's walk through an example
+with zlib. We first ask to analyze it. However, since we have more than one
+install, we are asked to disambiguate:
+
+.. code-block:: console
+
+    $ spack analyze zlib
+    ==> Error: zlib matches multiple packages.
+      Matching packages:
+        fz2bs56 zlib@1.2.11%gcc@7.5.0 arch=linux-ubuntu18.04-skylake
+        sl7m27m zlib@1.2.11%gcc@9.3.0 arch=linux-ubuntu20.04-skylake
+      Use a more specific spec.
+
+
+We can then specify the spec version that we want to analyze:
+
+.. code-block:: console
+
+    $ spack analyze zlib@1.2.11%gcc@7.5.0 arch=linux-ubuntu18.04-skylake
+
+If you don't provide any specific analyzer names, by default all analyzers 
+(shown in the ``--list-analyzers`` list) will be run. If an analyzer does not
+have any result, it will be skipped. For example, here is a result running for
+wget, which didn't have config args, and also did not have the libabigail
+analyzer implemented yet:
+
+.. code-block:: console
+
+    $ ls opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/wget-1.20.3-bum6zeaezbuqjhudzphcsyb4avkiizhj/.spack/analyze/
+    spack-analyzer-environment-variables.json  spack-analyzer-install-files.json
+
+If you want to run a specific analyzer, ask for it with `--analyzer`. Here we run
+spack analyze on libabigail (already installed) _using_ libabigail1
+
+.. code-block:: console
+
+    $ spack analyze --analyzer abigail libabigail
+
+You can now see the abigail results in the analyze folder!
+
+    $ ls opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/wget-1.20.3-bum6zeaezbuqjhudzphcsyb4avkiizhj/.spack/analyze/
+    spack-analyzer-environment-variables.json  spack-analyzer-install-files.json  spack-analyzer-libabigail-wget.xml
+
+It is currently flat (not compressed) xml, and will eventually be gzipped when libabigail supports it.
```

Comment:
We could, but if libabigail can't actually read the zipped content, we'd also need to unzip it as well. That would save us space in a database binary field over a full text, but require a bit of extra work (probably reasonable to do). I agree we should remove this sentence here since we don't really know, and update the docs if it eventually does output compressed.